smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/ReactiveKafkaProducer.java

```diff
@@ -136,6 +145,54 @@ public ReactiveKafkaProducer(Map<String, Object> kafkaConfiguration, String chan
         });
     }
 
+    public Uni<Void> initTransactions() {
+        return runOnSendingThread((Consumer<Producer<K, V>>) Producer::initTransactions);
+    }
+
+    @Override
+    public Uni<Void> beginTransaction() {
+        return runOnSendingThread((Consumer<Producer<K, V>>) Producer::beginTransaction);
+    }
+
+    @Override
+    public Uni<Void> commitTransaction() {
+        return runOnSendingThread((Consumer<Producer<K, V>>) Producer::commitTransaction);
+    }
+
+    @Override
+    public Uni<Void> abortTransaction() {
+        return runOnSendingThread((Consumer<Producer<K, V>>) Producer::abortTransaction);
+    }
+
+    @Override
+    public Uni<Void> sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offsets,
+            ConsumerGroupMetadata groupMetadata) {
+        return runOnSendingThread(producer -> {
+            producer.sendOffsetsToTransaction(offsets, groupMetadata);
+        });
+    }
+
+    @Override
+    public Uni<Void> withTransaction(Consumer<MultiEmitter<? super ProducerRecord<K, V>>> work) {
+        return beginTransaction()
+                .chain(() -> Multi.createFrom().<ProducerRecord<K, V>> emitter(multiEmitter -> {
+                    try {
+                        work.accept(multiEmitter);
+                    } catch (Throwable throwable) {
+                        multiEmitter.fail(throwable);
+                    } finally {
+                        multiEmitter.complete();
+                    }
+                })
+                        .onItem().transformToUniAndConcatenate(this::send)
+                        .collect().asList())
```

Comment:
What's the reason to collect the iteam?

Reply:
It was to ensure that all records from `MultiEmitter` were actually sent before committing the transaction. 
IIRC `toUni` did only took the first item.

BTW maybe should remove this method, though it is working, it was an early test I did and replaced by the `KafkaTransactionsImpl`.

Reply:
I thought the `ReactiveKafka{Consumer,Producer}` are supposed to expose the reactive analogues of the low-level Kafka API and anything higher-level (like this `withTransactions`) is supposed to live elsewhere? (In other words, I guess I agree with removing this, since the higher-level API people should actually use is `KafkaTransactions`.)

Reply:
I agree, removing this.